### DISCUSSION TITLE
Question about Newton-Raphson's reverse AD

### DISCUSSION THREAD

**Discussion by DoTulip** [jax-ml/jax#7898] [Q&A] [answered]
Hello JAX team, I have read through the tutorial part of the JAX document and am still have some questions about how AD is implemented in JAX. Does JAX support Newton-Raphson's reverse AD? If it is not currently supported, will it be developed in the future? @jakevdp

**Comment by soraros** [comment, marked as answer]
Yes. How to implement memory-efficient reverse AD for Newton-Raphson is explained in this [tutorial](http://implicit-layers-tutorial.org/implicit_functions/)/[talk](https://www.youtube.com/watch?v=MX1RJELWONc). This [library](https://github.com/google/jaxopt) provides a convenient API using the same techniques.

**Reply by DoTulip** [reply]
Thank you very much for your quick response! It is very helpful to me!
